server/channels/app/user.go

```diff
@@ -1784,6 +1784,57 @@ func (a *App) PermanentDeleteUser(rctx request.CTX, user *model.User) *model.App
 		return err
 	}
 
+	afterId := strings.Repeat("0", 26)
+	for {
+		channels, err := a.Srv().Store().Channel().GetGroupAndDirectChannelsForUser(user.Id, afterId, 1000, true)
+		if err != nil {
+			return model.NewAppError("PermanentDeleteUser", "app.channel.get_all_direct.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+
+		if len(channels) == 0 {
+			break
+		}
+
+		for _, channel := range channels {
+			afterId = channel.Id
+			var newCh *model.Channel
+			var appErr *model.AppError
```

Comment:
Looks like we are repeating some logic here. Maybe there's a chance for some refactoring?

Reply:
It's a bit convoluted on app migrations, I think it's better to let them stay separate.